api/lib/mno_enterprise/concerns/controllers/pages_controller.rb

```diff
@@ -23,15 +23,15 @@ module MnoEnterprise::Concerns::Controllers::PagesController
   # TODO: Access + existence checks could be added in the future. This is not
   # mandatory as Mno Enterprise will do it anyway
   def launch
-    app_instance = MnoEnterprise::AppInstance.find_by(uid: params[:id])
+    app_instance = Mno::AppInstance.where(uid: params[:id]).first
     MnoEnterprise::EventLogger.info('app_launch', current_user.id, 'App launched', app_instance)
     redirect_to MnoEnterprise.router.launch_url(params[:id], {wtk: MnoEnterprise.jwt(user_id: current_user.uid)}.reverse_merge(request.query_parameters))
   end
 
   # GET /loading/:id
   # Loading lounge - wait for an app to be online
   def loading
-    @app_instance = MnoEnterprise::AppInstance.where(uid: params[:id]).reload.first
+    @app_instance = Mno::AppInstance.find(uid: params[:id]).first
```

Comment:
`find` should not be used. It is either `where().first` when there is conditions or `find_one` to find by id.